pkg/os/temp/persistent.go

```diff
@@ -0,0 +1,42 @@
+package temp
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/windmilleng/wat/errors"
+)
+
+// An implementation of Dir more suitable
+// for directories that need to persist.
+type PersistentDir struct {
+	dir string
+}
+
+func NewPersistentDir(path string) (*PersistentDir, error) {
+	_, err := os.Stat(path)
+	if err == nil || !os.IsNotExist(err) {
+		return nil, fmt.Errorf("NewPersistentDir: dir already exists: %s", path)
+	}
+
+	err = os.Mkdir(path, 0777)
+	if err != nil {
+		return nil, errors.Propagatef(err, "NewPersistentDir failed to create %s", path)
```

Comment:
i think we can remove the errors.Propagatef here...it's not actually doing anything more than a normal fmt.Errorf would